### PR TITLE
Allow Generator to take version_added parameter

### PR DESCRIPTION
**Pull request by tejasbubane**
Other libraries like `rubocop-rspec` which rely on rubocop's generator
will be able to provide in their own version to be used in generated config. 

example: https://github.com/rubocop-hq/rubocop-rspec/pull/905

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [ ] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/cop/generator.rb [hunk 1]
```diff
@@ -132,11 +132,12 @@ def inject_require(root_file_path: 'lib/rubocop.rb')
         ).inject
       end
 
-      def inject_config(config_file_path: 'config/default.yml')
+      def inject_config(config_file_path: 'config/default.yml',
+                        version_added: bump_minor_version)
```
Comment (koic) [inline lib/rubocop/cop/generator.rb:136]: I think this API is also used in RuboCop core. It seems that the extended parameter expected to be sent by cop developer from terminal.

File: spec/rubocop/cop/generator_spec.rb [hunk 2]
```diff
@@ -280,6 +280,28 @@ def on_send(node)
...
+    context 'with version provided' do
+      it 'uses the provided version' do
+        expect(File).to receive(:write).with(path, <<~YAML)
+          Style/Alias:
+            Enabled: true
+
+          Style/FakeCop:
+            Description: 'TODO: Write a description of the cop.'
+            Enabled: pending
+            VersionAdded: '1.52.1'
```
Comment (koic) [inline spec/rubocop/cop/generator_spec.rb:293]: Patch version is unnecessary.

File: lib/rubocop/cop/generator.rb (hunk 1 above)
Comment (tejasbubane) [reply lib/rubocop/cop/generator.rb:136]: `new_cop` task does not use any other parameter except cop name: https://github.com/rubocop-hq/rubocop/blob/218f476531aa596b89d0227f30530786a8899a82/tasks/new_cop.rake#L20

**Comment by pirj** [comment]
To clarify: currently, if run from `rubocop-rspec`, `rake new_cop[RSpec/Abc]` generates the following in `rubocop-rspec`'s `config/default.yml`:
```yml
RSpec/Abc:
  Description: 'TODO: Write a description of the cop.'
  Enabled: pending
  VersionAdded: '0.83'
```

While RuboCop RSpec's current version is 1.38.1, and `VersionAdded` should be `1.38.2`.

**Comment by tejasbubane** [comment]
Not sure why circleci is failing here.

**Comment by pirj** [comment]
Seems to be a temporary failure, please re-run the build 🤞

**Comment by pirj** [comment]
Good to merge?

**Comment by koic** [comment]
I'll merge this one.

**Pull request merged**